src/orders/components/OrderTransactionRefundPage/utils.ts

```diff
@@ -130,31 +132,47 @@ const validateQty = ({
   return value;
 };
 
-export const handleQtyToRefundChange = ({
+export const handleLinesToRefundChange = ({
   data,
-  qtyToRefund,
+  linesToRefund,
   order,
   draftRefund,
   setValue,
 }: {
   data: DatagridChangeOpts;
-  qtyToRefund: QuantityToRefund[];
+  linesToRefund: LineToRefund[];
   order: OrderDetailsGrantRefundFragment | undefined | null;
   draftRefund: OrderDetailsGrantRefundFragment["grantedRefunds"][0] | undefined;
   setValue: UseFormSetValue<OrderTransactionRefundPageFormData>;
 }) => {
-  const unchangedQuantites = qtyToRefund.filter(
+  const unchangedLines = linesToRefund.filter(
     qty => qty.row !== data.currentUpdate?.row,
   );
 
   if (data.currentUpdate) {
```

Comment:
this entire `if` statements can be moved to the separated function that will return for you an input for `setValue` call. To calculate that input at first glance you need `data` and `linesToRefund`. Consider limiting the `data` to the only entries you need